returnn/util/file_cache.py

```diff
@@ -99,7 +104,16 @@ def get_file(self, src_filename: str) -> str:
         :return: cached file path (in the cache directory)
         """
         dst_filename = self._get_dst_filename(src_filename)
-        self._copy_file_if_needed(src_filename, dst_filename)
+        last_error = None
+        for num_try in range(1, self._num_tries + 1):
```

Comment:
So as said, I would not do this retrying at all currently. I would first try with the other suggestions.

Reply:
Ok I reduced the retry logic to out of disk space errors and then I'd leave it in for now. I also changed the lock to work on the directory level itself.

Reply:
But not sure. I think I would avoid the retry logic. I don't see a good reason for doing it. But I see many problems with it.